tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -442,18 +467,19 @@ impl<E> Deref for FormattedFields<E> {
 
 // === impl FmtLayer ===
 
-macro_rules! event_from_span {
-    ($event:ident = $id:ident, $span:ident, $($field:ident = $value:expr),*) => {
+macro_rules! with_event_from_span {
+    ($id:ident, $span:ident, $($field:literal = $value:expr),*, |$event:ident| $code:tt) => {
```

Comment:
Nit, take it or leave it: I'm assuming the reason we match fields as literals rather than as idents is because we want to allow dotted fields, which are not  identifiers? We _could_ continue using `stringify!` here if we matched fields as `expr` fragments instead of `literal`s:
```suggestion
    ($id:ident, $span:ident, $($field:expr = $value:expr),*, |$event:ident| $code:tt) => {
```

Reply:
`expr` covers a lot of strings we don’t want to generate :-)

Reply:
fair enough --- i hadn't realized that `literal` has been stable long enough to be MSRV-compliant for us!